mutiny-core/src/nodemanager.rs

```diff
@@ -631,30 +631,41 @@ impl<S: MutinyStorage> NodeManager<S> {
 
         let gossip_sync = Arc::new(gossip_sync);
 
-        // load lsp clients, if any
-        let lsp_clients: Vec<LspClient> = match c.lsp_url.clone() {
+        let mut lsp_configs: Vec<LspConfig> = match c.lsp_url.clone() {
             // check if string is some and not an empty string
             Some(lsp_urls) if !lsp_urls.is_empty() => {
-                let urls: Vec<&str> = lsp_urls.split(',').collect();
-
-                let futs = urls.into_iter().map(|url| LspClient::new(url.trim()));
-
-                let results = futures::future::join_all(futs).await;
-
-                results
-                    .into_iter()
-                    .flat_map(|res| match res {
-                        Ok(client) => Some(client),
-                        Err(e) => {
-                            log_warn!(logger, "Error starting up lsp client: {e}");
-                            None
-                        }
-                    })
-                    .collect()
+                lsp_urls.split(',').map(|url| LspConfig::Voltage(url.trim().to_string())).collect()
+
+                // let futs = urls.into_iter().map(|url| );
+
+                // let results = futures::future::join_all(futs).await;
+
+                // results
+                //     .into_iter()
+                //     .flat_map(|res| match res {
+                //         Ok(client) => Some(AnyLsp::Voltage(client)),
+                //         Err(e) => {
+                //             log_warn!(logger, "Error starting up lsp client: {e}");
+                //             None
+                //         }
+                //     })
+                //     .collect()
             }
             _ => Vec::new(),
         };
 
+        if let Some(lsp_connection_string) = c.lsp_connection_string.clone() {
+            if !lsp_connection_string.is_empty() {
+                let lsp_connection_strings: Vec<&str> = lsp_connection_string.split(',').collect();
+                let lsp_token = c.lsp_token.unwrap_or_default();
+                let lsp_tokens: Vec<&str> = lsp_token.split(',').collect();
+
+                for (index, connection_string) in lsp_connection_strings.into_iter().enumerate() {
+                    lsp_configs.push(LspConfig::LSPS(LspsConfig { connection_string: connection_string.to_string(), token: lsp_tokens.get(index).map(|t| t.to_string()) }));
```

Comment:
Don't think will work properly if you have say 3 lsps and you don't need a token for the 2nd. Not sure the best way to handle this.